specification/dartLangSpec.tex

```diff
@@ -5204,28 +5206,43 @@ \subsubsection{Correct Member Overrides}
   $F$ must then be a subtype of $F'$.
 
   \commentary{%
-  The subtype requirement ensures that argument list shapes
-  that are admissible for an invocation of a method with signature $m'$
-  are also admissible for an invocation of a method with signature $m$.
-  For instance, $m'$ may accept 2 or 3 positional arguments,
-  and $m$ may accept 1, 2, 3, or 4 positional arguments, but not vice versa.
-  This is a built-in property of the function type subtype rules.
-  %
-  Note that a member signature differs from
-  an underlying syntactic declaration $D$ in a class $C$.
-  In particular,
-  a parameter in a member signature has the modifier \COVARIANT{}
-  if and only if the parameter is covariant-by-declaration
-  (\ref{covariantParameters}),
-  and that may be the case due to declarations in a supertype of $C$,
-  so that modifier need not be present in $D$.
-  %
-  There is an additional potential compile-time error associated with
-  a parameter which is covariant-by-declaration
-  (\ref{instanceMethods}).
-  But we cannot cover that here as a property of member overrides,
-  because it is concerned with declarations in all superinterfaces,
-  indirect as well as direct.%
+    The subtype requirement ensures that argument list shapes
+    that are admissible for an invocation of a method with signature $m'$
+    are also admissible for an invocation of a method with signature $m$.
+    For instance, $m'$ may accept 2 or 3 positional arguments,
+    and $m$ may accept 1, 2, 3, or 4 positional arguments, but not vice versa.
+    This is a built-in property of the function type subtype rules.
+
+    Note that a member signature differs from
+    an underlying syntactic declaration $D$ in a class $C$.
+    In particular,
+    a parameter in a member signature has the modifier \COVARIANT{}
+    if and only if the parameter is covariant-by-declaration
+    (\ref{covariantParameters}),
+    and that may be the case due to declarations in a supertype of $C$,
+    so that modifier need not be present syntactically in $D$.
+  }
+
+  If $p'$ is a formal parameter in $m'$ that has the modifier \COVARIANT,
+  the corresponding parameter $p$ in $m$ must also have the modifier \COVARIANT,
+  unless the type of $p$ is a top type
+  (\ref{superBoundedTypes}).
+
+  \commentary{%
+    This can only occur in the case where $m$ is the signature of
```

Comment:
This is still a predicate, not a situation. The predicate does not say where it will be used.
(Also, try to avoid the unqualified "this".)

Maybe phrase this as:

  This requirement is relevant in the case where $m$ is the signature of a concrete
  member declaration inherited from a superclass where the parameter was not covariant,
  and $m'$ is the signature of an interface method, added in a subclass, 
  that $m$ is required to implement.

Reply:
Done.